lockfile/src/cyclonedx.rs

```diff
@@ -164,23 +164,37 @@ fn from_purl<T: Component>(component: &T) -> anyhow::Result<Package> {
 
 pub struct CycloneDX;
 
+impl CycloneDX {
+    fn process_components<T: Component>(components: Option<&[T]>) -> anyhow::Result<Vec<Package>> {
+        components.map_or(Ok(Vec::new()), |comp| {
+            let component_iter = filter_components(comp);
+            let mut packages = Vec::new();
+            for comp in component_iter {
+                match from_purl(comp) {
+                    Ok(pkg) => packages.push(pkg),
+                    Err(e) => {
+                        if e.is::<UnknownEcosystem>() {
+                            log::warn!("{:?}", e)
+                        } else {
+                            bail!(e)
+                        }
```

Comment:
Just to roll up all the suggestions above. Here is the end result.

```suggestion
                        if !e.is::<UnknownEcosystem>() {
                            return Err(e.into());
                        }
```

Reply:
Nah. I like the simplified version